include/mfmg/common/hierarchy.hpp

```diff
@@ -24,13 +24,28 @@
 #include <mfmg/cuda/cuda_mesh_evaluator.cuh>
 #endif
 
+#include <deal.II/base/timer.h>
+
 #include <boost/property_tree/ptree.hpp>
 
 #include <memory>
 #include <vector>
 
 namespace mfmg
 {
+void timer_enter_subsection(std::shared_ptr<dealii::TimerOutput> const &timer,
```

Comment:
Why did you chose to pass a smart pointer over a reference?
When would the pointer be null?

Reply:
The pointer will always be null except when running the `hierarchy_driver` to profile the code.